app/controllers/admin/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Admin
+  class CommentsController < Admin::BaseController
+    load_and_authorize_resource
+
+    def index
+      @comments = grouped_comments(accessible_ordered_comments)
+      @unread_comments = grouped_comments(accessible_ordered_comments.find_since_last_login(current_user))
+      @posted_comments = grouped_comments(accessible_ordered_comments.find_comments_by_user(current_user))
+    end
+
+    private
+
+    def accessible_ordered_comments
+      Comment.accessible_by(current_ability).joins('INNER JOIN events ON commentable_id = events.id').order('events.title', 'comments.created_at DESC')
```

Comment:
I think not, because comments do not have a relatioinship with events (I assume you meant joins(:events)?). There is commentable which in our case is only referring to Event model.